src/duckling/game/drawable/drawable-drawer.ts

```diff
@@ -0,0 +1,96 @@
+import {Graphics, Container, DisplayObject} from 'pixi.js';
+
+import {getPosition} from '../position/position-attribute';
+import {Entity} from '../../entitysystem/entity';
+import {Vector, degreesToRadians} from '../../math';
+import {drawEllipse, drawRectangle, rgbToHex} from '../../canvas/drawing/util';
+
+import {DrawableAttribute, getDrawableAttribute} from './drawable-attribute';
+import {Drawable, DrawableType} from './drawable';
+import {ShapeDrawable} from './shape-drawable';
+import {ContainerDrawable} from './container-drawable';
+import {ShapeType, Shape} from './shape';
+import {Circle} from './circle';
+import {Rectangle} from './rectangle';
+
+export function drawDrawableAttribute(entity : Entity) {
+    var positionAttribute = getPosition(entity);
+    var drawableAttribute = getDrawableAttribute(entity);
+    if (!positionAttribute || !drawableAttribute.topDrawable) {
+        return null;
+    }
+
+    var container = new Container();
+    var drawable = new DisplayObject();
+    drawable = drawDrawable(drawableAttribute.topDrawable);
+    container.addChild(drawable);
+    drawable.updateTransform();
+    container.addChild(drawDrawableBounds(drawable.getBounds(), drawableAttribute.topDrawable));
+    container.position.x = positionAttribute.position.x;
+    container.position.y = positionAttribute.position.y;
+    return container;
+}
+
+function drawDrawable(drawable : Drawable) : DisplayObject {
+    if (drawable.inactive) {
+        return new DisplayObject();
+    }
+
+    var drawableContainer = new Container();
+    switch (drawable.type) {
+        case DrawableType.Shape:
+            drawableContainer.addChild(drawShapeDrawable(drawable as ShapeDrawable));
+        case DrawableType.Container:
+            drawableContainer.addChild(drawContainerDrawable(drawable as ContainerDrawable));
+    }
+    applyDrawableProperties(drawable, drawableContainer);
+    return drawableContainer;
+}
+
+function drawDrawableBounds(bounds: PIXI.Rectangle, drawable : Drawable) : DisplayObject {
+    var graphics = new Graphics();
+    drawable.bounds = {
```

Comment:
I don't think drawables should have a .bounds property.  Eventually we can cache the bounds, but for now just do the expensive calculation whenever you need.